pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -1052,7 +1052,15 @@ func (cb *ClusterBuilder) buildUpstreamClusterTLSContext(opts *buildClusterOpts,
 		if opts.serviceRegistry == provider.External && len(tls.SubjectAltNames) == 0 {
 			tls.SubjectAltNames = opts.serviceAccounts
 		}
-		if tls.CredentialName != "" {
+		if tls.InsecureSkipVerify != nil && tls.InsecureSkipVerify.Value {
```

Comment:
Should we just make `InsecureSkipVerify` not allowed when CaCertificates or SubjectAltNames is set? In validation

Reply:
I have no problem with `InsecureSkipVerify` only taking effect when `VerifyCertificateAtClient` is in use

Reply:
After revisiting this, I am a bit more concerned about `InsecureSkipVerify` not doing what is expected from the name and how it is used in programs like curl. If you feel strongly about it only effecting `VerifyCertificateAtClient`, I would be happy to go ahead and make changes.

Reply:
I don't mean anything to do with VerifyCertificateAtClient, I mean:
```
tls:
  caCertificates: /foo/bar
  insecureSkipVerify: true
```

make that invalid in the webhook. There is no need to set the SAN/ca certs and then not verify - it just opens room for confusion that they are actually used?

Reply:
Thanks for the clarification. I'll work on getting that up. Just wanted to make sure it is fine that it doesn't do exactly what [the RFC](https://docs.google.com/document/d/100Jvwjr9KkURiF2HtiOtRlAON8TqgskUyLfT17_xoRM/edit#heading=h.xw1gqgyqs5b) specified.